specs/WebResourceRequested-CustomScheme.md

```diff
@@ -0,0 +1,318 @@
+  // This allows the WebView2 app to be able to handle
+  // WebResourceRequested event for requests with the specified scheme and
+  // be able to navigate the WebView2 to the custom scheme. Once the environment
+  // is created, the registrations are valid and immutable throughout the
+  // lifetime of the associated WebView2s' browser process and any WebView2
+  // environments sharing the browser process must be created with identical
+  // custom scheme registrations, otherwise the environment creation will fail.
+  // Any further attempts to register the same scheme will fail.
+  // The URIs of registered custom schemes will be treated similar to http URIs for their origins.
+  // They will have tuple origins for URIs with host and opaque origins for
+  // URIs without host as specified in https://html.spec.whatwg.org/multipage/origin.html
+  // Example:
+  // custom-scheme-with-host://hostname/path/to/resource has origin of custom-scheme-with-host://hostname
+  // custom-scheme-without-host:path/to/resource has origin of custom-scheme-without-host:path/to/resource
+  // For WebResourceRequested event, the request URIs and filter URIs with custom schemes
+  // will be normalized according to RFC3986.
+
+  // The name of the custom scheme to register.
+  [propget] HRESULT SchemeName([out, retval] LPCWSTR* schemeName);
+  [propput] HRESULT SchemeName([in] LPCWSTR value);
+
+  // Whether the scheme will be treated as a secure context like https.
+  [propget] HRESULT IsSecure([out, retval] BOOL* isSecure);
+  // Set if the scheme will be treated as a secure context.
+  [propput] HRESULT IsSecure([in] BOOL value);
+
+  // If `isCorsEnabled` is set, the scheme will be allowed for requests subject to
+  // CORS rules as specified in https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS.
+  // If it is not set, CORS requests with this scheme will be blocked.
+  [propget] HRESULT IsCorsEnabled([out, retval] BOOL* isCorsEnabled);
+  // Set CORS enabled for the scheme.
+  [propput] HRESULT IsCorsEnabled([in] BOOL value);
+
+  // List of origins that are allowed to use the scheme,
+  // subject to cross origin restrictions.
+  // Origins are specified as a string in the format of scheme://host:port.
+  // `*` wildcard can also be used on origins.
+  // For example, "http://*.example.com:80".
+  // Any origin from the custom scheme itself is allowed by default. If the
+  // array is empty, no other origin except origins from same scheme are allowed.
+  HRESULT GetAllowedOrigins([out] UINT32* allowedOriginsCount, [out] LPCWSTR** allowedOrigins);
+  // Set the list of origins that are allowed to use the scheme.
+  HRESULT SetAllowedOrigins([in] UINT32 allowedOriginsCount, [in] LPCWSTR* allowedOrigins);
+}
+
+// This is the ICoreWebView2EnvironmentOptions3 interface
+[uuid(ac52d13f-0d38-475a-9dca-876580d6793e), object, pointer_default(unique)]
+interface ICoreWebView2EnvironmentOptions3 : IUnknown {
+  /// List of custom scheme registrations.
```

Comment:
Need to discuss in comment, ownership and stuff regarding the schemeRegistrations array. I expect that like strings in COM this is caller owned and caller must use CoTaskMemFree to decallocate